--- php-pecl-haru-1.0.0/config.m4~	2008-11-19 09:48:57.000000000 +0100
+++ php-pecl-haru-1.0.0/config.m4	2008-12-27 03:16:27.259726875 +0100
@@ -19,7 +19,7 @@
   SEARCH_FOR="include/hpdf.h"
   if test "$PHP_HARU" = "yes"; then
 	AC_MSG_CHECKING([for Haru in default path])
-    for i in "$SEARCH_PATH"; do
+    for i in $SEARCH_PATH; do
       if test -r "$i/$SEARCH_FOR"; then
         HARU_DIR="$i"
         AC_MSG_RESULT(found in $i)
